src/main/scala/com.snowplowanalytics.iglu/ctl/commands/VerifyRedshift.scala

```diff
@@ -18,23 +15,31 @@ object VerifyRedshift {
     getAllSchemasFromServer(command.igluServerUrl, command.apiKey, httpClient)
       .leftMap(NonEmptyList.one)
       .map(groupSchemasToFamilies)
-      .map(_.flatMap(verifyFamily))
-      .map(prepareOutputMessage)
+      .map(_.map(verifyFamily))
+      .map(prepareOutputMessage(_, command.verbose))
   }
 
-  private def verifyFamily(schemaFamily: SchemaFamily):  Option[SchemaCriterion] = {
-    val recoveryModels = foldMapMergeRedshiftSchemas(schemaFamily).recoveryModels
-    if (recoveryModels.nonEmpty) {
-      val key = recoveryModels.head._1
-      SchemaCriterion(key.vendor, key.name, key.format, key.version.model).some
-    } else None
+  private def verifyFamily(schemaFamily: SchemaFamily): Map[SchemaKey, NonEmptyList[String]] = {
+    foldMapMergeRedshiftSchemas(schemaFamily)
+      .recoveryModels
+      .view
+      .mapValues(_.errorAsStrings)
+      .toMap
   }
 
-  private def prepareOutputMessage(schemaCriterions: List[SchemaCriterion]): List[String] = {
-    if (schemaCriterions.nonEmpty) {
-      schemaCriterions.map(_.asString)
-    } else {
-      List("No breaking changes detected")
+  private def prepareOutputMessage(verifiedFamilies: List[Map[SchemaKey, NonEmptyList[String]]], verbose: Boolean): List[String] = {
```

Comment:
Perhaps we could add somewhere a couple of tests for rendered output messages?

And because we pull schemas from iglu server...I think we could do something similar to [this](https://github.com/snowplow/igluctl/blob/02eac737ecaf02120978d2686235f59d651da0b2/src/test/scala/com/snowplowanalytics/iglu/ctl/commands/TableCheckITSpec.scala#L207) so very simply mock HTTP server. 

Just an idea, if it's too much work let's move on!

Reply:
Thanks for the recommendation, @pondzix ! I like it and I'll add it in a separate commit.